### PR TITLE
Implemented inflight requests tracking

### PR DESCRIPTION
**Pull request by AvdLee**
Build in the base provider, available for all reactive providers. Linked with issue https://github.com/Moya/Moya/pull/229

### CODE REVIEW

**Comment by ealeksandrov** [comment]
Thanks for PR!
Your fork is behind of upstream. How about rebasing it on top of latest `master`?

**Comment by AvdLee** [comment]
Good point, done that!

**Comment by ashfurrow** [comment]
Looks great! A few notes:

- Needs a changelog entry.
- Tests for the ReactiveCocoa provider are good, need corresponding tests for RxSwift, and ideally for just Moya itself. Maybe using the cancellation token?

I'm _really_ busy this week, hopefully others get the chance to review. Otherwise I'll try to look at it this weekend. Thanks!

**Comment by AvdLee** [comment]
@ashfurrow I've added these tests. However, I had to make the `CancellableWrapper` public to get to the cancellableToken. I'm not really happy with that change, but maybe this is a good start for someone else to continue.

I'm out of time today (end of the day in the Netherlands 👍 ) Will try to create a changelog entry later this week!

**Comment by ashfurrow** [comment]
We've just merged #479, another breaking change (it's nice to group them all together). Now I'm thinking about other things we should break...

**Comment by AvdLee** [comment]
Changelog entry is added

File: Source/Moya.swift [hunk 1]
```diff
@@ -261,12 +305,12 @@ public func convertResponseToResult(response: NSHTTPURLResponse?, data: NSData?,
     }
 }
 
-private struct CancellableWrapper: Cancellable {
-    var innerCancellable: CancellableToken? = nil
+public struct CancellableWrapper: Cancellable {
+    public var innerCancellable: CancellableToken? = nil
```
Comment (ashfurrow) [inline Source/Moya.swift]: I was thinking, if we make this `internal`, then we can use `@testable import Moya` in our unit tests and avoid making this `public`. Thoughts?

**Comment by AvdLee** [comment]
@ashfurrow good find! Done that. Should be all good now.

**Comment by ashfurrow** [comment]
Looks good! CI failure is intermittent and doesn't look related to the PR, merging!

**Comment by AvdLee** [comment]
@ashfurrow the release of 6.5.0 should include this commits as noted with the release commits. However, checking https://github.com/Moya/Moya/blob/6.5.0/Source/Moya.swift shows there's no trackInflights included, where the master right now does. 

Which base branch is used for the release of tag 6.5.0?

**Comment by ashfurrow** [comment]
6.5.0 was released off a separate branch with a diverged history from master; it was necessary to rewind to before some breaking changes to get the `queue` specification included in a point release (instead of in our upcoming major release, which will include the changes in this PR). Does that make sense?

**Comment by AvdLee** [comment]
It does, I was confused as I saw my commits included in the release, but missed my changes. Thanks!

**Comment by Rep2** [comment]
Is this feature available for Rx?

**Comment by ashfurrow** [comment]
It ought to be – is it now working as expected for you?

**Comment by Rep2** [comment]
Sorry I was using 6.5.0. Updating to 7.0.0 did the job.

**Comment by ashfurrow** [comment]
Glad you figured it out 👍

**Comment by AvdLee** [comment]
Just to make this clear, the tracking is implemented inside the `MoyaProvider`. The `RxMoyaProvider` and `ReactiveCocoaMoyaProvider` inherit from this base class and therefor inherit the inflight tracking functionality.

**Pull request merged**